agent/functional_tests/tests/functionaltests_windows_test.go

```diff
@@ -178,7 +178,54 @@ func taskIamRolesTest(networkMode string, agent *TestAgent, t *testing.T) {
 	if containerMetaData.State.ExitCode != 42 {
 		t.Fatalf("Container exit code non-zero: %v", containerMetaData.State.ExitCode)
 	}
+}
+
+func TestV3TaskEndpointDefaultNetworkMode(t *testing.T) {
```

Comment:
This looks like duplicate code, can you extract them to be a function in the functional_test.go, so that they can be used in both windows and linux test.

Reply:
It's doable, but it will be confusing as we have to pass in a lot of parameters, here are the differences between linux and windows (maybe we will have more in the future):
1. Linux has network setting, but windows doesn't, and start a awsvpc mode needs to call a different function.
2. Their aws logs prefix name is different.
3. The container names are different.

If you still feel strongly that we need to abstract them, I can do it, otherwise I'd prefer not.

Reply:
Instead of passing parameters to the common method, can we have them as constants in the respective files? 
I'd strongly recommend reusing test code as much as possible.